imagehash/__init__.py

```diff
@@ -116,10 +116,10 @@ def average_hash(image, hash_size=8):
 def phash(image, hash_size=32):
 	image = image.convert("L").resize((hash_size, hash_size), Image.ANTIALIAS)
 	pixels = numpy.array(image.getdata(), dtype=numpy.float).reshape((hash_size, hash_size))
-	dct = scipy.fftpack.dct(pixels)
-	dctlowfreq = dct[:8, 1:9]
-	avg = dctlowfreq.mean()
-	diff = dctlowfreq > avg
+	dct = scipy.fftpack.dct(scipy.fftpack.dct(pixels, axis=0), axis=1)
+	dctlowfreq = dct[:8, :8]
```

Comment:
I'm really thankful for this library and I'd like to sincerely use it since I would not be able to implement these algorithms myself.

I don't think, though, that the algorithm works that way. I'm really guessing here as this is beyond my mathematical capabilities but I believe the part ```python dctlowfreq = dct[:8, 1:9]``` might be implementing *"Reduce the DCT. While the DCT is 32x32, just keep the top-left 8x8. Those represent the lowest frequencies in the picture."* step. Anyway I'm starting to wonder if the implementation is correct at all.

I tried to download Alyson_Hannigan picture and apply your hashing functions on her and then displaying the hash in visual form like this:

```python
from PIL import Image
import imagehash
img  = Image.open('Alyson_Hannigan_200512.jpg')
hash = imagehash.average_hash(img) # change method here
img2 = Image.fromarray(hash.hash.astype('uint8')*255)
img2.resize((80,80)).show()
```
Comparing with the images of the hashes from the articles the only similar is actually the **average_hash** other hash images are completely different. Notice that hackerfactor hash images are gray scale, except for *average_hash*, while hashes produced by this library are boolean black and white only. Is it side effect of scaling for the blog? Is this correct? Why do these differ so much?

Thanks for the reply.